fortanix-vme/fortanix-vme-runner/src/lib.rs

```diff
@@ -317,10 +305,18 @@ impl Server {
             .cloned()
     }
 
-    fn add_connection(&self, runner_enclave: VsockStream<Std>, runner_remote: TcpStream, remote_name: String) {
-        let k = ConnectionKey::from_vsock_stream(&runner_enclave);
-        let info = Connection::new(runner_enclave, runner_remote, remote_name);
-        self.connections.write().unwrap().insert(k.clone(), Arc::new(Mutex::new(info)));
+    fn add_connection(self: Arc<Self>, mut runner_enclave: VsockStream<Std>, mut runner_remote: TcpStream, remote_name: String) -> Result<JoinHandle<()>, IoError> {
+        thread::Builder::new().spawn(move || {
+            let k = ConnectionKey::from_vsock_stream(&runner_enclave);
+            let connection = Arc::new(Connection::new(&runner_enclave, &runner_remote));
+            self.connections.write().unwrap().insert(k.clone(), connection.clone());
```

Comment:
Perhaps it would be better to add the connection to the hash map in the parent thread instead of in the proxying thread.Then we have the guarantee that the connection will have been added to the map when the `add_connection()` function returns.

Reply:
The `connection.clone()` call or even the `Arc` don't seem necessary with the current code. Perhaps they are in preparation of PLAT-367?